src/client-side-encryption/state_machine.ts

```diff
@@ -526,15 +526,20 @@ export class StateMachine {
     tlsOptions: ClientEncryptionTlsOptions,
     options: tls.ConnectionOptions
   ): Promise<void> {
-    if (tlsOptions.tlsCertificateKeyFile) {
-      const cert = await fs.readFile(tlsOptions.tlsCertificateKeyFile);
-      options.cert = options.key = cert;
-    }
-    if (tlsOptions.tlsCAFile) {
-      options.ca = await fs.readFile(tlsOptions.tlsCAFile);
-    }
-    if (tlsOptions.tlsCertificateKeyFilePassword) {
-      options.passphrase = tlsOptions.tlsCertificateKeyFilePassword;
+    // If a secureContext is provided, it takes precedence over the other options.
+    if (tlsOptions.secureContext) {
```

Comment:
should we be making an analogous change in the driver in, e.g., V7? we don't currently do any overrides for it in the main client logic (see `_connect()`)

Reply:
@durran Were you able to verify the behavior for what happens when we pass both secureContext and the standalone options to Node in these two cases: (1) secureContext has valid auth, standalone options have invalid auth and (2) secureContext has invalid auth, standalone options have valid auth? If in both cases the secureContext is used and the other options are ignored, we should be able to make the same change to the driver option handling right now for consistency. Otherwise please file the V7 ticket to do so.

Reply:
Following the discussion on slack, the team consensus is that we should not have the short-circuit logic to skip mongodb tls option reading, instead, we should always read them and map to the corresponding node options, then pass those node options through along with secureContext, if specified.